code/game/objects/structures/extinguisher.dm

```diff
@@ -5,13 +5,14 @@
 	icon_state = "extinguisher"
 	anchored = TRUE
 	density = FALSE
-	var/obj/item/tool/extinguisher/has_extinguisher = new/obj/item/tool/extinguisher
+	var/obj/item/tool/extinguisher/has_extinguisher
 	var/opened = 0
 	var/base_icon
 
 /obj/structure/extinguisher_cabinet/Initialize()
 	. = ..()
 	base_icon = initial(icon_state)
+	has_extinguisher = new /obj/item/tool/extinguisher()
```

Comment:
the extinguisher should probably start in the cabinet
below in attackby it is set via contents which does it if you put it in manually
(worth noting that you should use forceMove and not operate on contents, either..)